blackjax/mcmc/integrators.py

```diff
@@ -68,43 +213,21 @@ def velocity_verlet(
     a1 = 0
     b1 = 0.5
     a2 = 1 - 2 * a1
-
-    logdensity_and_grad_fn = jax.value_and_grad(logdensity_fn)
-    kinetic_energy_grad_fn = jax.grad(kinetic_energy_fn)
-
-    def one_step(state: IntegratorState, step_size: float) -> IntegratorState:
-        position, momentum, _, logdensity_grad = state
-
-        momentum = jax.tree_util.tree_map(
-            lambda momentum, logdensity_grad: momentum
-            + b1 * step_size * logdensity_grad,
-            momentum,
-            logdensity_grad,
-        )
-
-        kinetic_grad = kinetic_energy_grad_fn(momentum)
-        position = jax.tree_util.tree_map(
-            lambda position, kinetic_grad: position + a2 * step_size * kinetic_grad,
-            position,
-            kinetic_grad,
-        )
-
-        logdensity, logdensity_grad = logdensity_and_grad_fn(position)
-        momentum = jax.tree_util.tree_map(
-            lambda momentum, logdensity_grad: momentum
-            + b1 * step_size * logdensity_grad,
-            momentum,
-            logdensity_grad,
-        )
-
-        return IntegratorState(position, momentum, logdensity, logdensity_grad)
-
+    cofficients = [b1, a2, b1]
```

Comment:
I feel that it's (very slightly) cleaner for the coefficients list to see [b1,a2], and for `generalized_symplectic_integrator` to transform it by `coeff = coeff[:-1] + coeff[::,-1]`.

Reply:
Since the coefficients are pretty small (i.e., still fit in one line), I would say prefer to be explicit here.